drivers/adc/adc_nrfx_saadc.c

```diff
@@ -86,110 +149,99 @@ static int adc_nrfx_channel_setup(const struct device *dev,
 
 	switch (channel_cfg->acquisition_time) {
 	case ADC_ACQ_TIME(ADC_ACQ_TIME_MICROSECONDS, 3):
-		config.acq_time = NRF_SAADC_ACQTIME_3US;
+		ch_cfg->acq_time = NRF_SAADC_ACQTIME_3US;
 		break;
 	case ADC_ACQ_TIME(ADC_ACQ_TIME_MICROSECONDS, 5):
-		config.acq_time = NRF_SAADC_ACQTIME_5US;
+		ch_cfg->acq_time = NRF_SAADC_ACQTIME_5US;
 		break;
 	case ADC_ACQ_TIME_DEFAULT:
 	case ADC_ACQ_TIME(ADC_ACQ_TIME_MICROSECONDS, 10):
-		config.acq_time = NRF_SAADC_ACQTIME_10US;
+		ch_cfg->acq_time = NRF_SAADC_ACQTIME_10US;
 		break;
 	case ADC_ACQ_TIME(ADC_ACQ_TIME_MICROSECONDS, 15):
-		config.acq_time = NRF_SAADC_ACQTIME_15US;
+		ch_cfg->acq_time = NRF_SAADC_ACQTIME_15US;
 		break;
 	case ADC_ACQ_TIME(ADC_ACQ_TIME_MICROSECONDS, 20):
-		config.acq_time = NRF_SAADC_ACQTIME_20US;
+		ch_cfg->acq_time = NRF_SAADC_ACQTIME_20US;
 		break;
 	case ADC_ACQ_TIME(ADC_ACQ_TIME_MICROSECONDS, 40):
-		config.acq_time = NRF_SAADC_ACQTIME_40US;
+		ch_cfg->acq_time = NRF_SAADC_ACQTIME_40US;
 		break;
 	default:
 		LOG_ERR("Selected ADC acquisition time is not valid");
 		return -EINVAL;
 	}
 
-	config.mode = (channel_cfg->differential ?
+	ch_cfg->mode = (channel_cfg->differential ?
 		NRF_SAADC_MODE_DIFFERENTIAL : NRF_SAADC_MODE_SINGLE_ENDED);
 
-	/* Keep the channel disabled in hardware (set positive input to
-	 * NRF_SAADC_INPUT_DISABLED) until it is selected to be included
-	 * in a sampling sequence.
-	 */
-
-	nrf_saadc_channel_init(NRF_SAADC, channel_id, &config);
-	nrf_saadc_channel_input_set(NRF_SAADC,
-				    channel_id,
-				    NRF_SAADC_INPUT_DISABLED,
-				    channel_cfg->input_negative);
-
-	/* Store the positive input selection in a dedicated array,
-	 * to get it later when the channel is selected for a sampling
-	 * and to mark the channel as configured (ready to be selected).
-	 */
-	m_data.positive_inputs[channel_id] = channel_cfg->input_positive;
+	nrfx_saadc_channel_config(&cfg);
+	/* Mark channel as configured (reaady to be slected) for a sampling. */
+	m_data.configured[channel_id] = true;
 
 	return 0;
 }
 
 static void adc_context_start_sampling(struct adc_context *ctx)
 {
-	nrf_saadc_enable(NRF_SAADC);
+	nrfx_err_t err;
 
-	if (ctx->sequence.calibrate) {
```

Comment:
This might need an update in the nrfx driver because function `nrfx_saadc_offset_calibrate` will reset selected sampling mode in the driver. We can either update the nrfx driver or we can call here as well:
```
        nrfx_err = nrfx_saadc_simple_mode_set(sequence->channels,                                    
                                              resolution,                                            
                                              oversampling,                                          
                                              irq_nrfx_event_handler);
```
This will require keeping all function arguments as global variables.